internal/core/src/storage/ChunkCache.cpp

```diff
@@ -44,28 +49,17 @@ ChunkCache::Remove(const std::string& filepath) {
     columns_.erase(path);
 }
 
-void
-ChunkCache::Prefetch(const std::string& filepath) {
-    auto path = std::filesystem::path(path_prefix_) / filepath;
-    ColumnTable::const_accessor ca;
-    if (!columns_.find(ca, path)) {
-        return;
-    }
-    auto column = ca->second;
-    auto ok =
-        madvise(reinterpret_cast<void*>(const_cast<char*>(column->Data())),
-                column->ByteSize(),
-                read_ahead_policy_);
-    AssertInfo(ok == 0,
-               fmt::format("failed to madvise to the data file {}, err: {}",
-                           path.c_str(),
-                           strerror(errno)));
-}
-
 std::shared_ptr<ColumnBase>
 ChunkCache::Mmap(const std::filesystem::path& path,
                  const FieldDataPtr& field_data) {
-    std::unique_lock lck(mutex_);
+    MmapFileLocks::accessor acc;
+    if (!mmap_file_locks_.find(acc, path)) {
+        mmap_file_locks_.insert(
```

Comment:
the two lines are not atomic, two threads are at line 56 and they can't find the path, then they both insert new mutex, one of the thread's `acc` will point to a dangling object

Reply:
Once the `find` method returns, the access lasts until the accessor is released. Having access to an element can block other threads/access. So I belive it's fine. see also： https://oneapi-src.github.io/oneTBB/main/tbb_userguide/concurrent_hash_map.html